code/go/0chain.net/smartcontract/zcnsc/nodes.go

```diff
@@ -357,6 +360,38 @@ func NewUserNode(id string) *UserNode {
 	}
 }
 
+func (un *UserNode) AddMintNonce(nonce int64) {
+	un.MintNonces = append(un.MintNonces, nonce)
+}
+
+func (un *UserNode) AddBurnTicket(ethereumAddress string, hash string, nonce int64) error {
+	burnTicketRaw := entity.BurnTicket{Hash: hash, Nonce: nonce}
+	burnTicket, err := json.Marshal(burnTicketRaw)
+	if err != nil {
+		return err
+	}
+	un.BurnTickets[ethereumAddress] = append(un.BurnTickets[ethereumAddress], burnTicket)
+	return nil
+}
+
+func (un *UserNode) GetBurnTickets(ethereumAddress string) ([]entity.BurnTicket, error) {
+	var result []entity.BurnTicket
+
+	burnTickets, ok := un.BurnTickets[ethereumAddress]
+	if !ok {
+		return result, nil
+	}
+
+	for _, burnTicket := range burnTickets {
+		var dst entity.BurnTicket
+		if err := json.Unmarshal(burnTicket, &dst); err != nil {
+			return nil, err
+		}
+		result = append(result, dst)
+	}
+	return result, nil
```

Comment:
Let's sort the result slice to give consistent order, random map order could lead to inconsistent state if the return value was not processed correct.